modules/upgrade/partials/incompat-changes.adoc

```diff
@@ -0,0 +1,9 @@
+=== Review incompatible changes
+
+* Starting in version 24.2, when managing configuration properties using the AlterConfigs API directly, or the `kafka-configs.sh` shell script without incremental flags, Redpanda resets all unspecified values to the default values. This aligns more closely with the behavior in Apache Kafka. There is no change if you're managing your configuration with tools like `rpk`, Redpanda Console, Kubernetes, Helm, or Terraform. 
++
+NOTE: An exception is made for the `redpanda.remote.` topic properties, such as `redpanda.remote.delete`. The remote properties are intentionally sticky to avoid the possibility of disabling Tiered Storage.
```

Comment:
> The remote properties are intentionally sticky to avoid the possibility of disabling Tiered Storage.

I believe the danger is that disabling these may result in the data in S3 for those topics may get deleted, am I correct here @mattschumpert ?

Reply:
Disabling may or may not actually delete the data immediately (I dont think it will) but it can certainly lead to filling disks as the retention setting now gets applied to local storage,  and  to storing topic data on storage devices that are meant to be ephemeral.

@micheleRP 

I think we can just say  something like "These properties are intentionally sticky to avoid the possibility of _unintentionally_ disabling Tiered Storage for the topic, which can cause significant operational issues in clusters designed/sized to use Tiered Storage exclusively and where object storage is the only persistent/durable storage tier"

Reply:
Rephrased note with Matt: 
This does not pertain to the `redpanda.remote.` topic properties, such as `redpanda.remote.delete`. The remote properties are not reset to their defaults by the AlterConfigs API to avoid the possibility of unintentionally disabling Tiered Storage for a topic, which could cause significant operational issues for clusters designed to use Tiered Storage exclusively or with object storage as the only durable storage tier.